### PR TITLE
Add Sync class with a puzzle to implement it

### PR DESCRIPTION
**Pull request by Sammers21**
Related to #160. 

`Sync` class has been created with a puzzle to implement it.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#161](https://github.com/artipie/asto/issues/161) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/85c40f1a-e1e6-4a30-b5de-20a03c0c9620); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/asto/Sync.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * @since 0.19
+ * @checkstyle UnusedPrivateField (500 lines)
+ * @checkstyle SingularField (500 lines)
+ * @checkstyle NonStaticMethodCheck (500 lines)
+ * @todo #160:30min Implement Sync class.
+ *  This class currently is not implemented. This class should be implemented in a way to storage
+ *  synchronization would work properly.
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField", "PMD.AvoidDuplicateLiterals"})
+public class Sync {
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/Sync.java:42]: @Sammers21 I don't like this class name, it's correlating with Java synchronization domain.

**Comment by Sammers21** [comment]
@g4s8, I have reworked the design. Review it again, please.

File: src/main/java/com/artipie/asto/Copy.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        this.to = to;
+        this.keys = keys;
+    }
+
+    /**
+     * Copy specified keys from one storage, to another.
+     *
+     * @return When copy operation completes
+     */
+    public CompletableFuture<Void> perform() {
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/Copy.java:78]: @Sammers21 this class has functions semantic and can be replaced with a function: `Copy.perform(from, to, keys)`. Perform method sounds like a function call too. We need to separate object's state from behavior parameters, e.g. by using only source storage as a state but target storage as a parameter. Also, we need to change method name to describe the behavior of the object, e.g. `copy()`

Comment (Sammers21) [reply src/main/java/com/artipie/asto/Copy.java]: Done

**Comment by 0crat** [comment]
Job was finished in 4 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @g4s8[/z](https://www.0crat.com/u/g4s8)[/z](https://www.0crat.com/u/g4s8) didn't make CR comments

**Comment by Sammers21** [comment]
@g4s8 , can you release a new version?

**Comment by g4s8** [comment]
@Sammers21 but this class is not yet implemented, why you may want a release with it?

**Pull request merged**